src/prototype_room_my.js

```diff
@@ -432,10 +432,26 @@ Room.prototype.executeRoom = function() {
   this.handleTerminal();
   this.handleNukeAttack();
 
-  if (Game.time % 10 === 0) {
+  if(!this.memory.spawnTimer){
+    this.memory.spawnTimer = 0;
+  }
+  var harvesters = this.find(FIND_MY_CREEPS, {
+    filter: function(object) {
+      if (object.memory.role === 'harvester') {
+        return object.memory.base == room.name;
+      }
+      return false;
+    }
+  });
+  if(harvesters.length < 1){
     this.spawnCheckForCreate();
   }
-
+  if(this.memory.spawnTimer > 100 ||  this.energyAvailable == this.energyCapacityAvailable || (this.controller.level *250 < this.energyCapacityAvailable && Game.Time % 50 == 0 ) || this.memory.attackTimer > 15){
+    this.memory.spawnTimer = 0;
+    this.spawnCheckForCreate();
+  } else {
+    this.memory.spawnTimer = this.memory.spawnTimer + 1;
+  }
```

Comment:
it causes the spawn to run 100 ticks after the last time it ran.  The goal is for it to always run when full - but of we use a mod Game.time for the last fallback it could happen in the middle of a fill cycle. this way it will run in the event of a room failure - but it will not run randomly.  It is the ultimate failure case.